File: SharpLayers/SharpMap.Presentation.Web.SharpLayers/Controls/Overview/OverviewMap.cs
using System;
using System.Web.UI.WebControls;
using System.Web.UI;
using System.ComponentModel;
using System.ComponentModel.Design;
using AjaxControlToolkit;

[assembly: System.Web.UI.WebResource("SharpMap.Presentation.Web.SharpLayers.Controls.Overview.OverviewMap.js", "text/javascript")]

namespace SharpMap.Presentation.Web.SharpLayers.Controls.Overview
{
    [ClientScriptResource("SharpMap.Presentation.Web.SharpLayers.Controls.Overview.OverviewMap", "SharpMap.Presentation.Web.SharpLayers.Controls.Overview.OverviewMap.js")]
    [TargetControlType(typeof(Control))]
    public class OverviewMap : ToolBaseComponent<OverviewMapBuilderParams>
    {

        protected override string ScriptComponentName
        {
            get { return "SharpMap.Presentation.Web.SharpLayers.Controls.Overview.OverviewMap"; }
        }
    }
}

File: SharpLayers/SharpMap.Presentation.Web.SharpLayers/Controls/Overview/OverviewMap.js

Type.registerNamespace('SharpMap.Presentation.Web.SharpLayers.Controls.Overview');

SharpMap.Presentation.Web.SharpLayers.Controls.Overview.OverviewMap = function() {
    SharpMap.Presentation.Web.SharpLayers.Controls.Overview.OverviewMap.initializeBase(this);

}
SharpMap.Presentation.Web.SharpLayers.Controls.Overview.OverviewMap.prototype = {
    initialize: function() {
        SharpMap.Presentation.Web.SharpLayers.Controls.Overview.OverviewMap.callBaseMethod(this, 'initialize');
    },

    dispose: function() {
        SharpMap.Presentation.Web.SharpLayers.Controls.Overview.OverviewMap.callBaseMethod(this, 'dispose');
    },
    _toolBuilderDelegate: function() {
        var options = this.get_builderParams();
        options.div = typeof options.div == "string" ? $get(options.div) : options.div;
        var mapHost = this.get_targetMapHost();
        options.mapOptions = mapHost.get_builderParams();
        return new OpenLayers.Control.OverviewMap(options);
    }
}
SharpMap.Presentation.Web.SharpLayers.Controls.Overview.OverviewMap.registerClass('SharpMap.Presentation.Web.SharpLayers.Controls.Overview.OverviewMap', SharpMap.Presentation.Web.SharpLayers.Controls.ToolBaseComponent);
